### PR TITLE
Add support to list impls of protocol in `info` middleware

### PR DESCRIPTION
**Pull request by kapilreddy**
Use ns-imports of all namespaces to list all classes that
extend the protocol in question.

HACK - Modify classname to `map->{classname}`. The jump will only work for
defrecords at the moment.

https://github.com/clojure-emacs/cider-nrepl/issues/567

### CODE REVIEW

**Comment by bbatsov** [comment]
@kapilreddy Thanks for looking in this!

I think the right home for this code would be `orchard`, not `cider-nrepl`. Ideally `cider-nrepl` should be a very think wrapper around the orchard library, which does the actually heavy lifting.

**Comment by kapilreddy** [comment]
Makes sense. Will push the changes on orchard.

**Comment by bbatsov** [comment]
@kapilreddy 👍

**Comment by bbatsov** [comment]
@arichiardi I guess it's time to remind you of your promise. ;-)

**Comment by arichiardi** [comment]
Ah ah definitely thanks for the reminder 😅

File: src/cider/nrepl/middleware/info.clj [hunk 1]
```diff
@@ -86,7 +87,28 @@
...
+                  xs (map (fn [x]
+                            (let [xs (str/split (pr-str x)
+                                                #"\.")]
+                              (str (str/replace (str/join  "." (drop-last xs))
+                                                #"\_"
+                                                "-")
+                                   "/map->"
+                                   (last xs))))
+                          x)]
+              {:candidates (into {}
```
Comment (arichiardi) [inline src/cider/nrepl/middleware/info.clj]: is this returning `:candidates`? shouldn't it go in `compilment`?

Comment (bbatsov) [reply src/cider/nrepl/middleware/info.clj]: Those are different candidates. :-)

Comment (bbatsov) [reply src/cider/nrepl/middleware/info.clj]: I think we use the same key for several things here and there (I think it was used somewhere in `info` as well).

**Comment by bbatsov** [comment]
@arichiardi Any progress with this?

**Comment by kapilreddy** [comment]
@arichiardi Hey, Let me know if you need help moving this into Orchard and/or testing it.

**Comment by arichiardi** [comment]
I think I need help yes, my Clojure time has been reduced substantially 😒

**Comment by viebel** [comment]
@bbatsov @arichiardi What's the status of this issue?

**Comment by bbatsov** [comment]
It's waiting for a volunteer to drive it to the finish line.

**Comment by viebel** [comment]
@kapilreddy Would you like to volunteer?

**Comment by vemv** [comment]
Probably it doesn't make much sense to keep the PR open for yet another year, it's bit of a distraction when checking out our projects' status.

Volunteering welcome! (Probably at Orchard level next time)

**Pull request closed without merging**